packages/core/src/utils.js

```diff
@@ -512,3 +512,19 @@ export class DefaultMap extends Map {
     return super.get(key);
   };
 };
+
+export function compareObjectTypes(obj1, obj2) {
+  if (obj1 === obj2) return true; // Handles primitives
+  if (typeof obj1 !== 'object' || typeof obj2 !== 'object' || obj1 === null || obj2 === null) return false;
+
+  const keys1 = Object.keys(obj1);
+  const keys2 = Object.keys(obj2);
+
+  if (keys1.length !== keys2.length) return false;
+
+  for (const key of keys1) {
+    if (!keys2.includes(key) || !compareObjectTypes(obj1[key], obj2[key])) return false;
+  }
```

Comment:
```
  name: tag.name || null,
                width: tag.width || null,
                height: tag.height || null,
                'os-name': tag.osName || null,
                'os-version': tag.osVersion || null,
                orientation: tag.orientation || null,
                'browser-name': tag.browserName || null,
                'browser-version': tag.browserVersion || null,
                resolution: tag.resolution || null
```
These are the only tag values, does it needes recursive checking ?

Reply:
Tomorrow the tag can change and evolve, we need to ensure that cli is fail proof so that we don't miss cases.

Reply:
- Resolved it